app/src/main/java/com/github/sdpteam15/polyevents/fragments/LoginFragment.kt

```diff
@@ -76,15 +81,39 @@ class LoginFragment : Fragment() {
     }
 
     private fun firebaseAuthWithGoogle(idToken: String) {
-        //Get the credential back and instanciate FirebaseAuth object
+        //Get the credential back and instantiate FirebaseAuth object
         val credential = GoogleAuthProvider.getCredential(idToken, null)
         FirebaseAuth.getInstance().signInWithCredential(credential)
             .addOnCompleteListener(activity as Activity) { task ->
                 if (task.isSuccessful) {
-                    HelperFunctions.changeFragment(
-                        activity,
-                        MainActivity.fragments[R.id.id_fragment_profile]
-                    )
+                    addIfNotInDB()
+                } else {
+                    failedLogin.show()
+                }
+            }
+    }
+
+    private fun addIfNotInDB() {
```

Comment:
Maybe we could split this method into separate sub-methods for readability ?

Reply:
Yes, totally. It is done now